### PR TITLE
feat(tier4_autoware_utils): add geometry utils

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Add geometry util functions.
This is part of https://github.com/autowarefoundation/autoware.universe/pull/873.

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/873

## Tests performed

not add test code.
check pull_over module #873 can work.

## Notes for reviewers

some functions are similar. maybe not needed.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp [hunk 1]
```diff
@@ -38,6 +38,67 @@
 #include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
 #endif
 
+// TODO(wep21): Remove these apis
+//              after they are implemented in ros2 geometry2.
```
Comment (tkimura4) [inline common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp:42]: Did you copy this comment from somewhere?

**Review by tkimura4 — APPROVED** [review]
LGTM

This PR only moved some utility functions to tier4_autoware_utils from some packages (freespace_planning_algorithms, occulusion_spot, behavior_path_planner

This PR

File: common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp (hunk 1 above)
Comment (kosuke55) [reply common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp:42]: yes, just copy from somewhere. It is originally commented but someone may removed it.
If not needed I will remove it.

Comment (tkimura4) [reply common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp:42]: OK. It's fine as it is.

**Comment by kosuke55** [comment]
@tkimura4 @yukkysaito 
I added tests for new functions in https://github.com/autowarefoundation/autoware.universe/pull/1257/commits/0cf064dd4588d5376f36e663aa96b80a3fd86f70
I confirmed that the expected values are the same as the values calculated from another library ([skrobot](https://github.com/iory/scikit-robot)).
```
import numpy as np
from skrobot.coordinates import make_coords
from skrobot.coordinates.math import rpy2quaternion

pos = [2.0, 4.0, 6.0]
q = rpy2quaternion(np.deg2rad([30, 20, 10]))  # yaw pitch roll
p = make_coords(pos, q)

pos = [1.0, 2.0, 3.0]
q = rpy2quaternion(np.deg2rad([30, 30, 30]))
trans = make_coords(pos, q)

p_transed = trans.copy_coords().transform(p)
print(p_transed)
# <Coordinates 0x7f758df2a850 5.384 5.067 8.232 / 1.2 0.5 1.0>
print(p_transed.quaternion)  # w x y z
# [0.767046   0.24304508 0.42968035 0.4098101 ]

p_inverse_transed = trans.inverse_transformation().transform(p)
print(p_inverse_transed)
# <Coordinates 0x7f553e99c970 0.116 2.833 2.442 / 0.1 -0.2 -0.4>
print(p_inverse_transed.quaternion)  # w x y z
# [ 0.98106026 -0.17298739 -0.08189961  0.02980902]
```

**Comment by yukkysaito** [comment]
pls add test code

File: common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp [hunk 2]
```diff
@@ -735,6 +735,175 @@ TEST(geometry, transformPoint)
...
+  using tier4_autoware_utils::inverseTransformPoint;
+  using tier4_autoware_utils::inverseTransformPose;
+
+  geometry_msgs::msg::Pose pose_transform;
+  pose_transform.position.x = 1.0;
+  pose_transform.position.y = 2.0;
+  pose_transform.position.z = 3.0;
+  pose_transform.orientation = createQuaternionFromRPY(deg2rad(30), deg2rad(30), deg2rad(30));
+
+  // calc expected valuea
```
Comment (yukkysaito) [inline common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp:890]: typo?

Comment (kosuke55) [reply common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp]: yes, fixed in https://github.com/autowarefoundation/autoware.universe/pull/1257/commits/8d0da87019b4d85aba4416b99c8d8c92e0c8856f

**Review by yukkysaito — APPROVED** [review]
LGTM

**Comment by kosuke55** [comment]
test failed in humble. the values are almost equal to expected.
image

**Comment by kosuke55** [comment]
EXPECT_NEAR fixed test fails https://github.com/autowarefoundation/autoware.universe/pull/1257/commits/6104bef2781a2b3c6f484251b1658e1b96d1ca51, so merged.

**Pull request merged**